File: oodo_frontend/src/types/attendance.ts
export type AttendanceStatus = 'Present' | 'Absent' | 'Half Day' | 'Leave'

export interface Attendance {
  id: string
  employee_id: string
  date: string
  check_in?: string
  check_out?: string
  work_hours: number
  extra_hours: number
  status: AttendanceStatus
}

File: odoo_backend/apps/attendance/serializers/attendance.py
from rest_framework import serializers

from apps.attendance.models import Attendance


class AdminAttendanceCreateSerializer(serializers.Serializer):
    employee_id = serializers.UUIDField()
    date = serializers.DateField()
    check_in = serializers.DateTimeField(required=False, allow_null=True, default=None)
    check_out = serializers.DateTimeField(required=False, allow_null=True, default=None)
    status = serializers.ChoiceField(choices=Attendance.Status.choices, required=False)


class AdminAttendanceUpdateSerializer(serializers.Serializer):
    status = serializers.ChoiceField(choices=Attendance.Status.choices, required=False)
    session_id = serializers.UUIDField(required=False)
    check_in = serializers.DateTimeField(required=False, allow_null=True)
    check_out = serializers.DateTimeField(required=False, allow_null=True)